paasta_tools/tron_tools.py

```diff
@@ -219,36 +219,18 @@ def get_spark_paasta_cluster(self):
     def get_spark_paasta_pool(self):
         return self.config_dict.get("spark_paasta_pool", "batch")
 
-    def get_vault_token_file(self, system_paasta_config):
-        return self.config_dict.get(
-            "vault_token_file", system_paasta_config.get_vault_token_file()
-        )
-
     def get_env(self):
         env = super().get_env()
         spark_env = {}
         if self.get_executor() == "spark":
-            system_paasta_config = load_system_paasta_config()
             spark_ui_port = pick_random_port(
                 f"{self.get_service()}{self.get_instance()}".encode()
             )
             spark_env = get_mesos_spark_env(
                 spark_app_name=f"tron_spark_{self.get_service()}_{self.get_instance()}",
                 spark_ui_port=spark_ui_port,
                 mesos_leader=find_mesos_leader(self.get_spark_paasta_cluster()),
-                mesos_secret=load_mesos_secret_for_spark(
-                    secret_provider_name=system_paasta_config.get_secret_provider_name(),
-                    soa_dir=self.soa_dir,
-                    service_name=self.get_service(),
-                    cluster_name=self.get_cluster(),
-                    secret_provider_kwargs={
-                        "vault_cluster_config": system_paasta_config.get_vault_cluster_config(),
-                        "vault_auth_method": "token",
-                        "vault_token_file": self.get_vault_token_file(
-                            system_paasta_config
-                        ),
-                    },
-                ),
+                mesos_secret=SPARK_MESOS_SECRET_KEY,
```

Comment:
Just curious how spark driver container (mesos slave) would know to decrypt this env variable?

Reply:
the Mesos master scans the environment variables for things starting with `SECRET(` or `SHARED_SECRET(` and will decrypt them before starting up the executor.